test/jit/test_scriptmod_ann.py

```diff
@@ -0,0 +1,251 @@
+
+        with self.assertRaisesRegex(RuntimeError, "Tried to set "
+                                    "nonexistent attribute"):
+            with self.assertWarnsRegex(UserWarning, "doesn't support "
+                                       "instance-level annotations on "
+                                       "empty non-base types"):
+                torch.jit.script(M())
+
+    def test_annotated_empty_dict(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.x: Dict[str, int] = {}
+
+            def forward(self, x: Dict[str, int]):
+                self.x = x
+                return 1
+
+        with self.assertRaisesRegex(RuntimeError, "Tried to set "
+                                    "nonexistent attribute"):
+            with self.assertWarnsRegex(UserWarning, "doesn't support "
+                                       "instance-level annotations on "
+                                       "empty non-base types"):
+                torch.jit.script(M())
+
+    def test_annotated_empty_optional(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.x: Optional[str] = None
+
+            def forward(self, x: Optional[str]):
+                self.x = x
+                return 1
+
+        with self.assertRaisesRegex(RuntimeError, "Wrong type for "
+                                    "attribute assignment"):
+            with self.assertWarnsRegex(UserWarning, "doesn't support "
+                                       "instance-level annotations on "
+                                       "empty non-base types"):
+                torch.jit.script(M())
+
+    def test_annotated_with_jit_empty_list(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.x = torch.jit.annotate(List[int], [])
+
+            def forward(self, x: List[int]):
+                self.x = x
+                return 1
+
+        with self.assertRaisesRegex(RuntimeError, "Tried to set "
+                                    "nonexistent attribute"):
+            with self.assertWarnsRegex(UserWarning, "doesn't support "
+                                       "instance-level annotations on "
+                                       "empty non-base types"):
+                torch.jit.script(M())
+
+    def test_annotated_with_jit_empty_dict(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.x = torch.jit.annotate(Dict[str, int], {})
+
+            def forward(self, x: Dict[str, int]):
+                self.x = x
+                return 1
+
+        with self.assertRaisesRegex(RuntimeError, "Tried to set "
+                                    "nonexistent attribute"):
+            with self.assertWarnsRegex(UserWarning, "doesn't support "
```

Comment:
just a nit, but you might want to replace these 2 strings with a global
WARN_INSTANCE_ANNOT_EMPTY_NON_BASE_TYPE = ".." and then have the string only defined in one place?